superset-frontend/src/SqlLab/components/QueryTable/index.tsx

```diff
@@ -29,133 +28,156 @@ import Button from 'src/components/Button';
+      actions?.queryEditorSetSql({ id: query.sqlEditorId }, query.sql);
+    };
+
+    const openQueryInNewTab = (query: Record<string, any>) => {
+      actions?.cloneQueryToNewTab(query, true);
     };
 
-    const openQueryInNewTab = query => {
-      props.actions.cloneQueryToNewTab(query, true);
+    const openAsyncResults = (
+      query: Record<string, any>,
+      displayLimit: number,
+    ) => {
+      actions?.fetchQueryResults(query, displayLimit);
     };
 
-    const openAsyncResults = (query, displayLimit) => {
-      props.actions.fetchQueryResults(query, displayLimit);
+    const clearQueryResults = (query: Record<string, any>) => {
+      actions?.clearQueryResults(query);
     };
 
-    const clearQueryResults = query => {
-      props.actions.clearQueryResults(query);
+    const removeQuery = (query: Record<string, any>) => {
+      actions?.removeQuery(query);
     };
 
-    const removeQuery = query => {
-      props.actions.removeQuery(query);
+    const statusAttributes = {
+      success: {
+        config: {
+          icon: <Icons.Check iconColor={theme.colors.success.base} />,
+          label: t('Success'),
+        },
+      },
+      failed: {
+        config: {
+          icon: <Icons.XSmall iconColor={theme.colors.error.base} />,
+          label: t('Failed'),
+        },
+      },
+      stopped: {
+        config: {
+          icon: <Icons.XSmall iconColor={theme.colors.error.base} />,
+          label: t('Failed'),
+        },
+      },
+      running: {
+        config: {
+          icon: <Icons.Running iconColor={theme.colors.primary.base} />,
+          label: t('Running'),
+        },
+      },
+      fetching: {
+        config: {
+          icon: <Icons.Queued iconColor={theme.colors.primary.base} />,
+          label: t('fetching'),
+        },
+      },
+      timed_out: {
+        config: {
+          icon: <Icons.Offline iconColor={theme.colors.grayscale.light1} />,
+          label: t('Offline'),
+        },
+      },
+      scheduled: {
+        config: {
+          icon: <Icons.Queued iconColor={theme.colors.grayscale.base} />,
+          label: t('Scheduled'),
+        },
+      },
+      pending: {
+        config: {
+          icon: <Icons.Queued iconColor={theme.colors.grayscale.base} />,
+          label: t('Scheduled'),
+        },
+      },
+      error: {
+        config: {
+          icon: <Icons.Error iconColor={theme.colors.error.base} />,
+          label: t('Unknown Status'),
+        },
+      },
     };
 
-    return props.queries
+    return queries
       .map(query => {
-        const q = { ...query };
-        const status = statusAttributes[q.state] || statusAttributes.error;
+        // query's type is original Query; Shallow-copy of query, q's type is QueryTableQuery. So that prop, sql passed to another component will remain string, the type of original Query
+        const q: QueryTableQuery = { ...query };
+        let status: any;
```

Comment:
Could you explain why you changed this?

Reply:
q.state type is string | Object<string,any>, Type 'Record<string, any>' cannot be used as an index type. for statusAttributes[q.state]. Thus having an condition to filter out only when q.state type is string, then index works; The reason why q.state is string | Object<string,any>, is that q.state later assigned to be an element. Thank you ( I also added this as a comment above the codes)

Reply:
I think this may be easier if you use query as the original value and q as the new object that you are creating. So you'll have `query: Query` and `q: QueryTableValue`. `query.state` is a string and q.state is an object. 

and only bring in the properties that you don't overwrite, like:
const { state, ...q } = query;

Then you can do `status = statusAttributes[query.state] || statusAttributes.error;`
later you can assign q.state to an object.

Reply:
updated on line 185-190. Thank you!